charts/apisix/templates/daemonset.yaml

```diff
@@ -0,0 +1,201 @@
+  labels:
+    {{- include "apisix.labels" . | nindent 4 }}
+spec:
+  selector:
+    matchLabels:
+      {{- include "apisix.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      annotations:
+        checksum/config: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum }}
+        {{- with .Values.apisix.podAnnotations }}
+        {{- toYaml . | nindent 8 }}
+        {{- end }}
+      labels:
+        {{- include "apisix.selectorLabels" . | nindent 8 }}
+    spec:
+      {{- with .Values.global.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      containers:
+        - name: {{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.apisix.image.repository }}:{{ .Values.apisix.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.apisix.image.pullPolicy }}
+          ports:
+            - name: http
+              containerPort: {{ .Values.gateway.http.containerPort }}
+              protocol: TCP
+            - name: tls
+              containerPort: {{ .Values.gateway.tls.containerPort }}
+              protocol: TCP
+            {{- if .Values.admin.enabled }}
+            - name: admin
+              containerPort: {{ .Values.admin.port }}
+              protocol: TCP
+            {{- end }}
+            {{- if .Values.serviceMonitor.enabled }}
+            - name: prometheus
+              containerPort: {{ .Values.serviceMonitor.containerPort }}
+              protocol: TCP
+            {{- end }}
+            {{- if and .Values.gateway.stream.enabled (or (gt (len .Values.gateway.stream.tcp) 0) (gt (len .Values.gateway.stream.udp) 0)) }}
+            {{- with .Values.gateway.stream }}
+            {{- if (gt (len .tcp) 0) }}
+            {{- range $index, $port := .tcp }}
+            - name: proxy-tcp-{{ $index | toString }}
+              containerPort: {{ $port }}
+              protocol: TCP
+            {{- end }}
+            {{- end }}
+            {{- if (gt (len .udp) 0) }}
+            {{- range $index, $port := .udp }}
+            - name: proxy-udp-{{ $index | toString }}
+              containerPort: {{ $port }}
+              protocol: UDP
+            {{- end }}
+            {{- end }}
+            {{- end }}
+            {{- end }}
+          readinessProbe:
+            failureThreshold: 6
+            initialDelaySeconds: 10
+            periodSeconds: 10
+            successThreshold: 1
+            tcpSocket:
+              port: {{ .Values.gateway.http.containerPort }}
+            timeoutSeconds: 1
+          lifecycle:
+            preStop:
+              exec:
+                command:
+                - /bin/sh
+                - -c
+                - "sleep 30"
+          env:
+            - name: INTRANET_IP
```

Comment:
Does `DaemonSet` need this? `Deployment` doesn't have that.

Reply:
enable `skywalking` feature, diff instance.
i add `Deployment` env attriubte

Reply:
@PGDream Sorry, cannot understand what you expressed. Could you describe it more elaborately?

Reply:
After enabling skywalking, if you use an APISIX cluster, you need to distinguish which instance the link information comes from

Reply:
But who uses this environment variable?

Reply:
skywalking user,check doamin flow

Reply:
But I can't find any use of this environment variable on either `skywalking-logger` plugin or `skywalking` plugin.

Reply:
apisix dashboard controller panel  can  skywalking plugin  config `INTRANET_IP` env var

Reply:
@PGDream Could you give us a link about this?

Reply:
sorry, `INTRANET_IP` env with `Prometheus`
apisix config.yaml: 
```
  prometheus:
    export_addr:
      ip: "${{INTRANET_IP}}"
      port: 9092
```

Reply:
I don't think this environment variable is a necessary option.  If you need it in your scenario, you can add it after this [PR](https://github.com/apache/apisix-helm-chart/pull/232) is merged

Reply:
OK

Reply:
@tokers  please review